### PR TITLE
Fix Bug 1205858, adds new default contact us tab to contact page

### PR DESCRIPTION
**Pull request by schalkneethling**
@jpetto r?

@flodolo I am pretty sure this is not localized but, just putting an l10n label on this in case. If indeed it is not needed, you can just remove the label. Thanks!

### CODE REVIEW

**Comment by flodolo** [comment]
Confirming, not localized.

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        <li><a href="https://support.mozilla.org/">{{ _('I’m having problems with using Firefox')}}</a></li>
+        <li><a href="https://sendto.mozilla.org/page/contribute/givenow-seq?preset=2&ref=Donation_Form&utm_campaign=Donation_Form&utm_source=mozilla.org&utm_medium=referral&utm_content=contact_us_page">{{ _('I want to donate to Mozilla')}}</a></li>
+        <li><a href="https://www.mozilla.org/foundation/trademarks/policy/">{{ _('I have questions about copyright trademarks and Mozilla')}}</a></li>
+        <li><a href="mailto:research_requests@mozilla.com">{{ _('I need information for an academic paper')}}</a></li>
+        <li><a href="https://wiki.mozilla.org/People:MozSpaces_Guidelines">{{ _('I want to hold an event in a Mozilla space')}}</a></li>
+        <li><a href="https://bugzilla.mozilla.org/form.dev-engagement-event">{{ _('I want Mozilla to sponsor my event')}}</a></li>
+        <li><a href="https://www.mozilla.org/styleguide/identity/mozilla/branding/">{{ _('I have a question about using Mozilla logos')}}</a></li>
+        <li><a href="#press_enquiry" id="press_enquiry_toggle">{{ _('I’m a member of the Press and have a question for Mozilla') }}</a></li>
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: Should this `<li>` be removed until the press contact form is completed ([bug 1210426](https://bugzilla.mozilla.org/show_bug.cgi?id=1210426))?

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  <div id="entry-container">
+    <section class="entry contact-us" id="contact" data-tab="contact">
+      <header>
+        <h2>{{ _('Contacting Mozilla')}}</h2>
+      </header>
+
+      <ul>
+        <li><a href="https://support.mozilla.org/">{{ _('I’m having problems with using Firefox')}}</a></li>
+        <li><a href="https://sendto.mozilla.org/page/contribute/givenow-seq?preset=2&ref=Donation_Form&utm_campaign=Donation_Form&utm_source=mozilla.org&utm_medium=referral&utm_content=contact_us_page">{{ _('I want to donate to Mozilla')}}</a></li>
+        <li><a href="https://www.mozilla.org/foundation/trademarks/policy/">{{ _('I have questions about copyright trademarks and Mozilla')}}</a></li>
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: Use the `url` helper here?

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        <li><a href="mailto:research_requests@mozilla.com">{{ _('I need information for an academic paper')}}</a></li>
+        <li><a href="https://wiki.mozilla.org/People:MozSpaces_Guidelines">{{ _('I want to hold an event in a Mozilla space')}}</a></li>
+        <li><a href="https://bugzilla.mozilla.org/form.dev-engagement-event">{{ _('I want Mozilla to sponsor my event')}}</a></li>
+        <li><a href="https://www.mozilla.org/styleguide/identity/mozilla/branding/">{{ _('I have a question about using Mozilla logos')}}</a></li>
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: And here?

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        with any of our products, including Firefox for desktop and Android, Firefox OS
+        and Thunderbird, please <a href="{{ link_sumo }}">visit our support site</a>.
+      {% endtrans %}
+      </p>
+
+      <div class="contact-online">
+        <h4>{{ _('Find us online') }}</h4>
+        <ul>
+          <li><a href="https://facebook.com/mozilla" hreflang="en-US">{{ _('Facebook') }}</a></li>
+          <li><a href="https://twitter.com/mozilla" hreflang="en-US">{{ _('Twitter') }}</a></li>
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: Put a `rel="external"` on these two social links?

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          <li><a href="/about/forums/">{{ _('Forums &amp; e-mail lists') }}</a></li>
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: Use `url` helper here?

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html (hunk 1 above)
Comment (schalkneethling) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: Indeed, completely neglected to remove it.

File: bedrock/mozorg/templates/mozorg/contact/contact-landing.html (hunk 5 above)
Comment (schalkneethling) [inline bedrock/mozorg/templates/mozorg/contact/contact-landing.html]: Yeah, good point. This was all copied from preexisting code but, no reason not to improve it ;)

File: media/js/mozorg/contact-spaces.js [hunk 6]
```diff
@@ -115,12 +115,16 @@
             //set the current tab navigation item
             $('ul.category-tabs li[data-id="' + tab + '"]').addClass('current');
 
-            // set the current list menu navigation item
-            $('.nav-category li[data-id="' + id + '"]').addClass('current');
+            if (tab === 'spaces') {
+                // set the current list menu navigation item for the active space
+                $('.nav-category li[data-id="' + id + '"]').addClass('current');
```
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: I believe this line is still needed for both `spaces` and `communities`, as both have side navs.

File: media/js/mozorg/contact-spaces.js [hunk 7]
```diff
@@ -115,12 +115,16 @@
...
+            }
 
-            // hide the community sub menu's
-            $('.accordion .submenu').hide();
-            mozMap.toggleCommunitySubMenu();
+            if (tab === 'communities') {
+                // hide the community sub menu's
+                $('.accordion .submenu').hide();
```
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: I know the logic isn't changed in this PR, but I feel this line would be clearer if targeted at the `nav-communities` ID, as it appears the intent is to affect a specific element and not all `.accordion` elements. Thoughts?

Also, looks like a conditional around these two `communities` related lines should have been there before this PR, as they shouldn't need to fire for `spaces` or `contact`. :+1:

File: media/js/mozorg/contact-spaces.js [hunk 8]
```diff
@@ -778,6 +805,30 @@
         },
 
         /*
+         * Show the contact us page content.
+         * Determined using data-id attribute and .current list item.
+         */
+        showContactUsContent: function(url, cacheId) {
+            var $current = $('.category-tabs li.current');
+            var id = $current.data('id');
```
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: Doesn't look like this local variable is used.

File: media/js/mozorg/contact-spaces.js [hunk 9]
```diff
@@ -778,6 +805,30 @@
...
+            var contentUrl = url || current.prop('href');
```
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: Missing `$` in front of `current`.

File: media/css/mozorg/contact-spaces.less [hunk 10]
```diff
@@ -437,6 +437,11 @@
     }
 }
 
+/* override float rule for contact us tab */
+.entry.contact-us {
+    float: none;
```
Comment (jpetto) [inline media/css/mozorg/contact-spaces.less]: Should include the `.clearfix()` mixin here, as contains floated content?

**Comment by alexgibson** [comment]
Before merging this it might be a good idea to double-check mcom-tests to see if anything needs updating :+1:

**Comment by jpetto** [comment]
Looks like there's a bug on mobile, as the select boxes for both Spaces and Communities are visible when toggling back to the "Contact" tab.

screen shot 2015-10-08 at 1 39 32 pm

Also, desktop & tablet views looks a *little* bit off due to the amount of white space on the right side of the content. Since there's no side nav, perhaps we should remove the width restriction for the Contact content?

File: media/css/mozorg/contact-spaces.less (hunk 10 above)
Comment (schalkneethling) [inline media/css/mozorg/contact-spaces.less]: Did not run into any floating issues without the mixin so, perhaps it is not required?

**Comment by schalkneethling** [comment]
@jpetto addressed all comments. @alexgibson so, this one will be blocked until the mcom tests are updated?

This, for example, will need to be updated:
https://github.com/mozilla/mcom-tests/blob/master/tests/test_contact.py#L29

**Comment by alexgibson** [comment]
I think I might have spotted a history bug in this change, but will double-check soon to see if I can reproduce (currently mobile). Yes I think we'll likely need to update mcom-tests for this change too.

**Comment by jpetto** [comment]
Hm, found another layout issue - tabs are unevenly spaced at mobile wide:

screen shot 2015-10-08 at 3 37 25 pm

Could probably get rid of that top border for Contact mobile view as well, as there's no select box up top to separate.

File: media/css/mozorg/contact-spaces.less (hunk 10 above)
Comment (jpetto) [inline media/css/mozorg/contact-spaces.less]: The `.clearfix` on the parent `.container` is saving us from layout issues. If we had content underneath "Questions & feedback" however, there would be issues. I tend to err on the side of clearing floats. Doesn't hurt anything. :wink:

File: media/js/mozorg/contact-spaces.js (hunk 6 above)
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: Still need to remove the conditional around this line?

Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: Ah, I see it's repeated below. Would be easier/more DRY to just have the line once, not wrapped in a conditional though, yeah?

File: media/js/mozorg/contact-spaces.js [hunk 11]
```diff
@@ -115,12 +115,17 @@
...
+                // set the current list menu navigation item for the active space
+                $('.nav-category li[data-id="' + id + '"]').addClass('current');
+            }
 
-            // hide the community sub menu's
-            $('.accordion .submenu').hide();
-            mozMap.toggleCommunitySubMenu();
+            if (tab === 'communities') {
+                // hide the community sub menu's
+                $('#nav-communities').hide();
```
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: Ah, but this still has to target the child `.submenu`'s. Should have been more clear in my previous comment.

**Comment by jpetto** [comment]
Jeez, just noticed the tabs are unevenly spaced at desktop size as well. So subtle (and noisy with map bg) that I didn't notice it at first.

screen shot 2015-10-08 at 4 13 55 pm

File: media/js/mozorg/contact-spaces.js [hunk 12]
```diff
@@ -311,6 +322,8 @@
                 $('#nav-spaces, #meta-spaces').fadeOut(100, function () {
                     $('#nav-communities, #meta-communities').show();
                 });
+            } else if (tab === 'contact') {
+                $('#nav-spaces, #meta-spaces, #nav-communities, #meta-communities').fadeOut(100);
```
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: For consistency, let's keep the logic uniform and make the condition here `(tab === 'contact' && wideMode)`. Then we can simplify/ the `toggleMobileNav` function below to be:

```
} else if (tab === 'contact' && !wideMode) {
    $('#nav-communities-select, #meta-communities, #nav-spaces-select, #meta-spaces').hide();
}
```

**Comment by jpetto** [comment]
I'm seeing a history bug - possibly the one @alexgibson noticed. Load `/contact/`, click the "Spaces" tab, wait for content to load, click back button. The content does not switch back to "Contact". If you go from "Spaces" to "Communities" (or vice versa), the content updates correctly using the back/forward buttons.

**Comment by alexgibson** [comment]
> I'm seeing a history bug - possibly the one @alexgibson noticed. Load /contact/, click the "Spaces" tab, wait for content to load, click back button. The content does not switch back to "Contact". If you go from "Spaces" to "Communities" (or vice versa), the content updates correctly using the back/forward buttons.

Yep this is what I'm seeing too. Specifically, I think the new code needs to handle [this condition](https://github.com/mozilla/bedrock/blob/master/media/js/mozorg/contact-spaces.js#L160-L161)

**Comment by schalkneethling** [comment]
Ok, PR opened against mcom tests https://github.com/mozilla/mcom-tests/pull/443

File: media/js/mozorg/contact-spaces.js (hunk 6 above)
Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: :+1:

**Comment by alexgibson** [comment]
I did a bit of debugging locally on the history bug and I think these areas may still need updating:

https://github.com/mozilla/bedrock/blob/master/media/js/mozorg/contact-spaces.js#L534-L538
https://github.com/mozilla/bedrock/blob/master/media/js/mozorg/contact-spaces.js#L395

Also, we have a much [simplified version](https://github.com/mozilla/bedrock/blob/master/media/js/mozorg/contact-spaces-ie7.js) for older IE browsers. Does this need updating at all for this new tab?

File: media/js/mozorg/contact-spaces.js (hunk 11 above)
Comment (jpetto) [inline media/js/mozorg/contact-spaces.js]: Just a reminder to switch this to `$('#nav-communities .submenu').hide();` in the next update.

**Comment by schalkneethling** [comment]
@jpetto test all the things but, I need to update the IE bits that @alexgibson mentioned above. Was not even aware of that. Thanks Alex

**Comment by jpetto** [comment]
The history bug seems to be fixed. :+1: 

However, during testing, I noticed something odd. On any initial page load (`/contact/`, `/contact/spaces/`, `/contact/communities/`), the JS wipes out the main content, then an XHR request is made requesting the exact same page, which re-populates the main content. Essentially, it looks like the mechanism for smoothly swapping content when clicking tabs is operating at page load as well, resulting in a totally unnecessary request/performance hit.

Check the network tab in dev tools and you'll see an XHR request to the same page initially loaded. :confounded: 

I believe these three functions are all culprits:

`showSpace`
`showCommunityContent`
`showContactUsContent`

This behavior doesn't look to be introduced with this PR, but it sure would be nice to fix it up soon. Could be in this PR, or file a new bug if other matters are pressing.

**Comment by alexgibson** [comment]
@jpetto - nice spot. Not sure when this crept in, but you're right we should fix it. Probably just need to add a check to make sure the ajax doesn't get called on page load :+1:

**Comment by schalkneethling** [comment]
Thanks @jpetto I will resolve that in this PR as well.

**Comment by schalkneethling** [comment]
@jpetto updated. Let me know if there are other items you run into. Thanks!

**Comment by jpetto** [comment]
The last thing I see is that the initial content is not cached, which can cause an extra XHR request. If I load `/contact/` and click the "Spaces" tab, an XHR request is made to get the "Spaces" content. All good there. If I then click the back button or the "Contact" tab, another XHR request is made to get the "Contact" content (even though that content was retrieved on initial page load).

Maybe we could factor out the [caching mechanism from the AJAX request](https://github.com/mozilla/bedrock/blob/master/media/js/mozorg/contact-spaces.js#L1078-L1084) and use it in the [`setInitialContentState`](https://github.com/mozilla/bedrock/blob/master/media/js/mozorg/contact-spaces.js#L395) function to cache initially loaded content?

**Comment by schalkneethling** [comment]
@jpetto So, it seems the cacheid is not being set, let me look into it. Will update the PR and let you know. Thanks!

**Comment by schalkneethling** [comment]
@jpetto yes, so as you mentioned and I just confirmed, the cacheId is not set and hence the ajax request happens. Now the problem is, on initial load there are no ajax requests as the appropriate page is loaded directly.

I guess the way to get this to work is to change the way the page works overall in that the content is always loaded via Ajax, even on initial load i.e. the same page is loaded for `/contact/`, `/contact/spaces/` and 'contact/communities/` but, based on the URL, the appropriate content is loaded via Ajax thus, setting the cacheId from initial load.

This should probably be part of the refactor that is on the cards for the very near future. Perhaps there is a stop gap I am overlooking that we can implement in this bug but else, it is probably out of scope for this bug. Thoughts? Thanks!

**Comment by jpetto** [comment]
I *think* the fix could be in factoring out the caching mechanism from the AJAX request, then using it on page load *and* AJAX requests. From what I've seen of the code, AJAX isn't required to populate the cache, but they are tightly coupled at the moment.

However, I feel the page is at a good state in this PR. The redundant XHR request at page load has been removed, which is a big win (and was also outside the scope of this bug). In the interest of getting this new tab out the door, I'd be fine with opening another bug to finish up polishing the caching performance.

Just needs a squash and we'll be good to go. r+! :rocket:

**Comment by jpetto** [comment]
Also, just noting that [this pre-existing rewrite rule](https://github.com/mozilla/bedrock/blob/master/etc/httpd/global.conf#L298) should handle this new page:

`RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?contact(.*)$ /b/$1contact$2 [PT]`

**Comment by schalkneethling** [comment]
Awesome, thx for the review @jpetto Squashed. With regards to the rewrite rule yes, it will work just fine. There was a redirect that would redirect requests to `spaces/` but is removed here.

So, let's merge this first. I will then merge master onto the inquiry page, add the link and we can merge that one.

**Pull request merged**